### PR TITLE
[User] Documentation + return objects cleanup

### PR DESCRIPTION
**Pull request by gabrielsaruhashi**
Swagger documentation + response object cleanup
image
image

### CODE REVIEW

File: src/authentication/authentication.controller.ts [hunk 1]
```diff
@@ -1,62 +1,103 @@
...
+import ResponseDtoService from '@src/common/services/ResponseDtoService';
 import RegisterDto from './dto/register.dto';
 import { InmateAuthenticationGuard } from './guards/inmateAuthentication.guard';
 import { UserAuthenticationGuard } from './guards/userAuthentication.guard';
-import RequestWithInmate from './interfaces/requestWithInmate.interface';
+import RequestWithInmate from './interfaces/requestWithInmate.dto';
+
+// TODO: figure out a better way to document the DTOs for authentication endpoints
+// This is currently only used for Swagger documentation to circumvent the lack of dtos of th Strategy files
+class LoginDto {
```
Comment (lspdrz) [inline src/authentication/authentication.controller.ts:28]: Should we put in a dto directory?

File: src/authentication/guards/userAuthentication.guard.ts [hunk 2]
```diff
@@ -7,6 +7,7 @@ import { AuthGuard } from '@nestjs/passport';
 @Injectable()
 export class UserAuthenticationGuard extends AuthGuard('user') {
   async canActivate(context: ExecutionContext) {
+    console.log('here');
```
Comment (lspdrz) [inline src/authentication/guards/userAuthentication.guard.ts:10]: remove?

File: src/authentication/strategies/user.strategy.ts [hunk 3]
```diff
@@ -12,7 +11,8 @@ export default class UserStrategy extends PassportStrategy(Strategy, 'user') {
       usernameField: 'email',
     });
   }
-  async validate(email: string, password: string): Promise<PublicUserRo> {
+  async validate(email: string, password: string): Promise<UserRO> {
+    console.log('validating user sstrategy');
```
Comment (lspdrz) [inline src/authentication/strategies/user.strategy.ts:15]: remove?

File: src/common/interfaces/responses.dto.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+// connection returned to free world users
+export class UserConnectionRO {
+  id: number;
+  userId: number;
+  inmateId: number;
+  status: ConnectionStatus;
+  statusDetails: string;
+  relationship: ConnectionRelationship;
+  createdAt: Date;
+  inmate: InmateRO;
```
Comment (mpekala23) [inline src/common/interfaces/responses.dto.ts:36]: perfect

File: src/users/dto/addConnection.dto.ts [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import ConnectionRelationship from '@src/connections/enums/connectionRelationship.enum';
-import { IsNumber, IsEnum } from 'class-validator';
+import { IsNumber, IsEnum, } from 'class-validator';
```
Comment (mpekala23) [inline src/users/dto/addConnection.dto.ts:2]: can get rid of trailing ,?

**Review by mpekala23 — APPROVED** [review]
lgtm!

**Pull request merged**